src/lib/pubkey/hss_lms/hss.h

```diff
@@ -0,0 +1,392 @@
+/**
+ * HSS - Hierarchical Signatures System (RFC 8554)
+ * (C) 2023 Jack Lloyd
+ *     2023 Fabian Albert, Philippe Lieser - Rohde & Schwarz Cybersecurity GmbH
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#ifndef BOTAN_HSS_H_
+#define BOTAN_HSS_H_
+
+#include <botan/asn1_obj.h>
+#include <botan/rng.h>
+#include <botan/internal/int_utils.h>
+#include <botan/internal/lms.h>
+
+#include <memory>
+#include <span>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+namespace Botan {
+
+/**
+ * @brief The index of a node within a specific LMS tree layer
+ */
+using HSS_Sig_Idx = Strong<uint64_t, struct HSS_Sig_Idx_, EnableArithmeticWithPlainNumber>;
```

Comment:
Without a strong opinion: In many other algorithms we centralized the existing strong types into a `algo_types.h`. I did find this quite useful because it provides a first overview of the artefacts involved.

Reply:
We consciously decided against it to keep the hierarchy of LMS' building blocks separate.